src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Attributes/CheckCanApproveNoteAttributeTests.cs

```diff
@@ -0,0 +1,187 @@
+﻿namespace EA.Weee.Web.Tests.Unit.Areas.Scheme.Attributes
+{
+    using System;
+    using System.Web.Mvc;
+    using System.Web.Routing;
+    using AutoFixture;
+    using Core.AatfEvidence;
+    using Core.Scheme;
+    using Core.Shared;
+    using EA.Weee.Api.Client;
+    using EA.Weee.Tests.Core;
+    using EA.Weee.Web.Areas.Scheme.Attributes;
+    using EA.Weee.Web.Services.Caching;
+    using FakeItEasy;
+    using FluentAssertions;
+    using Weee.Requests.AatfEvidence;
+    using Weee.Tests.Core.DataHelpers;
+    using Xunit;
+
+    public class CheckCanApproveNoteAttributeTests : SimpleUnitTestBase
+    {
+        private readonly CheckCanApproveNoteAttribute attribute;
+        private readonly ActionExecutingContext context;
+        private readonly IWeeeClient client;
+        private readonly IWeeeCache cache;
+        private readonly Guid pcsId;
+        private readonly Guid evidenceNoteId;
+
+        public CheckCanApproveNoteAttributeTests()
+        {
+            client = A.Fake<IWeeeClient>();
+            cache = A.Fake<IWeeeCache>();
+            attribute = new CheckCanApproveNoteAttribute { Cache = cache, Client = () => client };
+            context = A.Fake<ActionExecutingContext>();
+            
+            pcsId = TestFixture.Create<Guid>();
+            evidenceNoteId = TestFixture.Create<Guid>();
+
+            var routeData = new RouteData();
+            routeData.Values.Add("pcsId", pcsId);
+            routeData.Values.Add("evidenceNoteId", evidenceNoteId);
+
+            A.CallTo(() => context.RouteData).Returns(routeData);
+        }
+
+        [Fact]
+        public void CheckCanApproveNoteAttribute_ShouldBeDerivedFromCheckTransferNoteAttributeBaseAttribute()
```

Comment:
i guess the name needs to be update to ShouldBeDerivedFromCheckSchemeNoteAttributeBase